Source/Layout/ASLayout.mm

```diff
@@ -216,52 +200,70 @@ - (void)setRetainSublayoutLayoutElements:(BOOL)retainSublayoutLayoutElements
 
 #pragma mark - Layout Flattening
 
+- (BOOL)isFlattened
+{
+  // A layout is flattened if its position is null, and all of its sublayouts are of type displaynode with no sublayouts.
+  if (! ASPointIsNull(_position)) {
+    return NO;
+  }
+  
+  for (ASLayout *sublayout in _sublayouts) {
```

Comment:
Would `NSFastEnumeration` allow us to cast that as `unowned void *` and cast down with a bridge to the type we need? In fact does it do a retain for the sub layouts in here?

Reply:
Fast enumeration doesn't retain the objects.